scripts/checks/compareGasReports.js

```diff
@@ -44,7 +44,25 @@ class Report {
     if (JSON.stringify(update.config.metadata) !== JSON.stringify(ref.config.metadata)) {
       throw new Error('Reports produced with non matching metadata');
     }
-    return Object.keys(update.info.methods)
+
+    const deployments = update.info.deployments
+      .map(contract => Object.assign(
+        contract,
+        { previousVersion: ref.info.deployments.find(({ name }) => name === contract.name) },
+      ))
+      .filter(contract => contract.gasData?.length && contract.previousVersion?.gasData?.length)
+      .flatMap(contract => [{
+        contract: contract.name,
+        method: '[bytecode length]',
+        avg: variation(contract.bytecode.length - 2, contract.previousVersion.bytecode.length - 2),
+      }, {
+        contract: contract.name,
+        method: '[construction cost]',
+        avg: variation(...[contract.gasData, contract.previousVersion.gasData].map(x => ~~average(...x))),
+      }])
+      .sort((a, b) => `${a.contract}:${a.method}` < `${b.contract}:${b.method}` ? -1 : 1);
```

Comment:
Minor issue for the purposes of this script, but the comparator function is expected to be symmetric, which this one isn't. On the other hand, maybe in this case we know that the array doesn't have duplicates? In any case a simple alternative is to use `str1.localeCompare(str2)`.